### PR TITLE
add new feature to Template module

### PR DESCRIPTION
**Pull request by alejandrojnm**
- You can now list all templates, create a new one, update and delete any templates that belonged to the user

### CODE REVIEW

File: template.go [hunk 1]
```diff
@@ -4,36 +4,97 @@ import (
...
+		return nil, err
+	}
+
+	return templates, nil
+}
+
+// NewTemplate this function will create a new template for the current user
+func (c *Client) NewTemplate(conf *Template) (*SimpleResponse, error) {
+	if conf.ImageID == "" {
+		if conf.VolumeID == "" {
```
Comment (andyjeffries) [inline template.go]: This should be a single `if conf.ImageID == "" && conf.VolumeID == "" {`

File: template.go [hunk 2]
```diff
@@ -4,36 +4,97 @@ import (
...
-	err = json.NewDecoder(bytes.NewReader(resp)).Decode(&templates)
-	for _, template := range templates {
+	if err := json.NewDecoder(bytes.NewReader(resp)).Decode(&templates); err != nil {
+		return nil, err
+	}
+
+	return templates, nil
+}
+
+// NewTemplate this function will create a new template for the current user
```
Comment (andyjeffries) [inline template.go]: `// NewTemplate will create a new template for the current user`

No need for a "this function"

File: template.go [hunk 3]
```diff
@@ -4,36 +4,97 @@ import (
...
 	ShortDescription string `json:"short_description"`
 	Description      string `json:"description"`
 	DefaultUsername  string `json:"default_username"`
 	CloudConfig      string `json:"cloud_config"`
 }
 
-// GetTemplateByCode finds the Template for an account with the specified code
-func (c *Client) GetTemplateByCode(code string) (*Template, error) {
+// ListTemplates return all template in system
+func (c *Client) ListTemplates() ([]Template, error) {
```
Comment (andyjeffries) [inline template.go]: Why have you replaced `GetTemplateByCode`? Having a `listTemplates` method is good, but the other function is a useful one still.

File: template.go [hunk 4]
```diff
@@ -4,36 +4,97 @@ import (
...
+
+	resp, err := c.SendPostRequest("/v2/templates", conf)
+	if err != nil {
+		return nil, err
+	}
+
+	return c.DecodeSimpleResponse(resp)
+}
+
+// UpdateTemplate this function will update a template for the current user
```
Comment (andyjeffries) [inline template.go]: Again remove `this function`, it should read like a sentence.

File: template.go [hunk 5]
```diff
@@ -4,36 +4,97 @@ import (
...
+func (c *Client) UpdateTemplate(id string, conf *Template) (*Template, error) {
+	if conf.ImageID == "" {
+		if conf.VolumeID == "" {
```
Comment (andyjeffries) [inline template.go]: As above, single `if` please.

File: template.go (hunk 3 above)
Comment (andyjeffries) [reply template.go]: Never mind, I see it's just diff weirdness - it's further down the page.

File: template.go (hunk 1 above)
Comment (alejandrojnm) [reply template.go]: Yes but in this case `&&` is `and` the doc in the API say this 

> This is the ID of a bootable volume, either owned by you or global (optional; but must be specified if no template_id is specified).

so i verify first if `ImageID` is present, if not i check `VolumeID` if not, i rise the error

Comment (alejandrojnm) [reply template.go]: On second thought if we could use `&&`

**Pull request merged**